plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1554,6 +1558,31 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         }
     }
 
+    @Override
+    public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, io.trino.spi.type.Type type)
```

Comment:
Is this guaranteed to be a base level column? Or could it be a field of a row type

Reply:
It could be any type. Such type is blocked in Iceberg side though.
```java
  @Override
  public UpdateSchema updateColumn(String name, Type.PrimitiveType newType) {
...
    Preconditions.checkArgument(
        TypeUtil.isPromotionAllowed(field.type(), newType),
        "Cannot change column type: %s: %s -> %s",
        name,
        field.type(),
        newType);
```

```java
  public static boolean isPromotionAllowed(Type from, Type.PrimitiveType to) {
    // Warning! Before changing this function, make sure that the type change doesn't introduce
    // compatibility problems in partitioning.
    if (from.equals(to)) {
      return true;
    }

    switch (from.typeId()) {
      case INTEGER:
        return to == Types.LongType.get();

      case FLOAT:
        return to == Types.DoubleType.get();

      case DECIMAL:
        Types.DecimalType fromDecimal = (Types.DecimalType) from;
        if (to.typeId() != Type.TypeID.DECIMAL) {
          return false;
        }

        Types.DecimalType toDecimal = (Types.DecimalType) to;
        return fromDecimal.scale() == toDecimal.scale()
            && fromDecimal.precision() <= toDecimal.precision();
    }

    return false;
  }
```

Reply:
Gotcha. 

I'd wouldn't expect dereference pushdown to be run for these types of operations, but in general as a rule I try to assume IcebergColumnHandles may be dereferenced columns. Mind just adding a `verify(column.isBaseColumn(), ...)`?

Reply:
Or switch the `column.getName()` call below to `column.getQualifiedName()`

Reply:
Added `verify(column.isBaseColumn(), "Cannot change nested field types");`